src/core/polyhedral/memory_promotion_heuristic.cc

```diff
@@ -503,5 +545,114 @@ void promoteGreedilyAtDepth(
   mapCopiesToThreads(mscop, unrollCopies);
 }
 
+namespace {
+isl::val getParamValIfFixed(isl::union_set uset, int pos) {
```

Comment:
This one feels reminiscent of a function that @abadams was asking for and that @skimo-openhub exposed for us from ISL. Maybe I am remembering wrong but maybe there is a way to avoid writing this function at all?
If we do need to write it indeed, it looks like a good candidate to move to ```include/tc/external/detail/islpp.h```.

Reply:
it uses the function that @skimo-openhub mentioned, but that function is only working for _sets_ not unions of sets; so I had to write a wrapper